packages/block-editor/src/components/list-view/use-list-view-client-ids.js

```diff
@@ -10,37 +10,26 @@ import { useSelect } from '@wordpress/data';
 import { store as blockEditorStore } from '../../store';
 import { unlock } from '../../lock-unlock';
 
+const NON_DISABLED_EDITING_MODES = [ 'default', 'contentOnly' ];
```

Comment:
The downside of this that once a new editing mode is added, you'll need to update this list. Not sure how realistic this risk is. A good solution would be to have two list-like arguments, allow-list and deny-list, and passing a `[ 'disabled' ]` denylist.

Reply:
You're right!

My first instinct was to have it so that you pass a predicate function.

```js
getClientIdsTreeWithBlockEditingMode( ( mode ) => mode !== 'disabled' )
```

Then I wondered why make the selector so specific to block editing mode? It could accept a query-like object that lets you do whatever filtering you want.

```js
getClientIdsTree( { rootBlockId: 'foo', blockEditingMode: ( mode ) => mode !== 'disabled' } )
```

Then I started to think I'm over-engineering this and YAGNI 😅 

The selector is private and only used by List View so instead I think let's emphasise that fact by moving the `!== 'disabled'` logic into the selector and renaming it to mention `ListView` by name. I've done that in 2ff27b069b.

That keeps things simple and doesn't prevent us from doing something generic if needed in the future.